src/bhfmm/containers/UniformPseudoParticleContainer.cpp

```diff
@@ -4319,10 +4314,10 @@ void UniformPseudoParticleContainer::communicateHalosOverlapPostProcessingStart(
 #endif
 }
 
-void UniformPseudoParticleContainer::communicateHalosAlongAxis(double *lowerNeighbourBuffer,
-															   double *higherNeighbourBuffer,
-															   double *lowerNeighbourBufferRec,
-															   double *higherNeighbourBufferRec,
+void UniformPseudoParticleContainer::communicateHalosAlongAxis(std::vector<double>lowerNeighbourBuffer,
+															   std::vector<double>higherNeighbourBuffer,
+															   std::vector<double>lowerNeighbourBufferRec,
+															   std::vector<double>higherNeighbourBufferRec,
```

Comment:
```suggestion
void UniformPseudoParticleContainer::communicateHalosAlongAxis(std::vector<double> &lowerNeighbourBuffer,
															   std::vector<double> &higherNeighbourBuffer,
															   std::vector<double> &lowerNeighbourBufferRec,
															   std::vector<double> &higherNeighbourBufferRec,
```
Ok here we need to pass non `const` references because MPI has no serious concept of `const` and we want to propagate the changes to the inputs to the outside.